rows.go

```diff
@@ -326,23 +326,30 @@ func (x *XlsxFile) parseRow(decoder *xml.Decoder, startElement *xml.StartElement
 // to interpret the value of any of the cells.
 func (x *XlsxFile) parseRawCells(rawCells []rawCell, index int) ([]Cell, error) {
 	cells := []Cell{}
-	for _, rawCell := range rawCells {
+
+	for i, rawCell := range rawCells {
+		ok := false
 		if rawCell.Value == nil && rawCell.InlineString == nil {
-			// This cell is empty, so ignore it
-			continue
+			if (x._maxColumnToKeepEmptyVals == 0) || (x._maxColumnToKeepEmptyVals < i) {
```

Comment:
It doesn't look like anything is setting `_maxColumnToKeepEmptyVals` other than the unit test you've added - so I'm a bit confused what the logic here is for.

Reply:
The unit test I added is based off the report for https://github.com/TheDataShed/xlsxreader/issues/40

If the user passes the number of columns like in the test, he will be
able to keep empty cells so that each array length is the same regardless
of whether there is data in it or not. I didn't want to change
the arguments all the way down the chain so I just set it as a config
on the reader that he can set to 81 so he gets the empty values that
he wants.

Reply:
I think the issue for #40 is that the reader skips over cells that are omitted from the XML markup, and doesn't fill them in. 
You can have a sparsely populated spreadsheet (i.e. the XML only contains info for columns 1 and 3, and omits info entirely for column 2). 

I think the 'correct' thing to do might be to have a utility that re-populates spare info, but it's not essential to the library, so has been omitted thus-far.